### PR TITLE
Random cleanups

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Here's a bunch of cleanups inspired by the recent coverity scanning.  The last one is pretty big, and will require a very close reading.

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
One thing I forgot to mention in the commit message:

While tidying, in `cockpitinternalmetrics.c`, I completely removed the code for parsing of the `instances` option, because nothing is using it.  That's an actual functional change, because it means that we won't throw the error anymore if we receive it incorrectly.

I'm not sure if we need to worry about that change or not, but in any case, it definitely needs a mention in a commit message.

File: src/bridge/cockpitinternalmetrics.c [hunk 1]
```diff
@@ -527,9 +518,6 @@ cockpit_internal_metrics_finalize (GObject *object)
 {
   CockpitInternalMetrics *self = COCKPIT_INTERNAL_METRICS (object);
 
-  g_free (self->instances);
-  g_free (self->omit_instances);
```
Comment (allisonkarlitskaya) [inline src/bridge/cockpitinternalmetrics.c:531]: This removal is incorrect.  I'll put that back.

Comment (allisonkarlitskaya) [reply src/bridge/cockpitinternalmetrics.c:531]: Nice to see that the valgrind tests caught this too!

Fixed in the latest push.

File: src/bridge/cockpithttpstream.c [hunk 2]
```diff
@@ -418,23 +417,14 @@ relay_headers (CockpitHttpStream *self,
...
+      g_autoptr(GBytes) message = cockpit_json_write_bytes (object);
       cockpit_channel_send (channel, message, TRUE);
-      g_bytes_unref (message);
     }
   else
     {
       cockpit_channel_control (channel, "response", object);
     }
 
-  json_object_unref (object);
```
Comment (marusak) [inline src/bridge/cockpithttpstream.c:430]: duMM question: `json_object_unref`, `g_bytes_unref` and `g_has_table_unref` are the `free` methods for `JsonObject`, `GBytes` and `GHashTable` respectively. So if using `g_autoptr` it is smart enough to know how to properly free these things up?

Comment (marusak) [reply src/bridge/cockpithttpstream.c:430]: neverMindMe: https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#g-autoptr :)

File: src/bridge/cockpitrouter.c [hunk 3]
```diff
@@ -615,19 +615,17 @@ strv_to_json_array (gchar **strv)
...
-  gint length;
   gint i;
 
   g_assert (config_args != NULL);
   g_assert (key != NULL);
 
   arr = g_ptr_array_new ();
-  length = g_strv_length (config_args);
-  for (i = 0; i < length; i++)
+  for (i = 0; config_args[i]; i++)
```
Comment (marusak) [inline src/bridge/cockpitrouter.c:628]: So this one breaks the loop when `config_args[i]` is `null`, right? Is it given that `config_args` cannot contain anything else, that would be "falsy"?

**Review by marusak — CHANGES_REQUESTED** [review]
Nice, I did the best review I could with my C skills and all seems fine to me :)
I like those cleanups with auto pointers and removing of `goto`s!

Needs closes though, but tests seem happy as well.

File: src/bridge/cockpitrouter.c (hunk 3 above)
Comment (allisonkarlitskaya) [reply src/bridge/cockpitrouter.c:628]: The only value of a pointer that evaluates false is NULL.

Comment (allisonkarlitskaya) [reply src/bridge/cockpitrouter.c:628]: Worth noting the implementation of `g_strv_length()` itself:

```
guint
g_strv_length (gchar **str_array)
{
  guint i = 0;

  g_return_val_if_fail (str_array != NULL, 0);

  while (str_array[i])
    ++i;

  return i;
}
```

Comment (marusak) [reply src/bridge/cockpitrouter.c:628]: makes sense, thanks!

**Comment by marusak** [comment]
meh, unit tests fail miserably.  This is due to npm being down/unreachable/out of limits...
In the previous push they were green and since then you just added closes line + rebased on master.
I am certain this is fine to land.

**Review by marusak — APPROVED** [review]
Thanks!

**Pull request merged**